expose and document run_data and scan_data on nodes (#347)
chef/automate: components/automate-chef-io/content/docs/nodes.md

When a user adds a node integration, like aws or azure, nodes are added to the `/nodes` endpoint.

> When an inspec report or chef client run is ingested, a node is added to the `/nodes` endpoint. If it already exists, the last contact time is updated, along with run data and scan data, that stores the latest run id or report id and status (as well as the penultimate status).

Reviewer: ```suggestion
When an inspec report or chef client run is ingested, a node is added to the `/nodes` endpoint. If the node already exists, its last contact time, run data, and scan data are updated, storing the status and the latest run id or report id, and the penultimate status.
```

That last sentence has a lot going on. This is my attempt to clean it up. WDYT?

> When an inspec report or chef client run is ingested, a node is added to the `/nodes` endpoint. If it already exists, the last contact time is updated, along with run data and scan data, that stores the latest run id or report id and status (as well as the penultimate status).

Reviewer 2: How is this? 
```suggestion
When an InSpec report or a Chef Infra Client run is ingested, a node is added to the `/nodes` endpoint. If the node already exists, its last contact time, run data, and scan data are updated. When the run data and scan data are updated, the latest information is stored for the run id or report id, the status, and the penultimate status.
```

- region
- source_id (a reference to the primary provider's node)
- state

> - statechange_timerange (supports two timestamps of type "2019-03-05T00:00:00Z")

Reviewer: Does the timestamp conform to any standard, eg ISO8601?

Author: urr..good question.

Reviewer 3: RFC3339 is the date time standard that Chef tries to use.

- status
- tags
- last_run_timerange (last time reported on ingested ccr: supports two timestamps of type "2019-03-05T00:00:00Z")
- last_scan_timerange (last time reported on ingested scan: supports two timestamps of type "2019-03-05T00:00:00Z")
- last_run_status (status on last ingested ccr)
- last_scan_status (status on last ingested scan)
- last_run_penultimate_status

> - last_scan_penultimate_status

Reviewer 3: Can we add a comment here for the `penultimate` fields? I had to look up what 'penultimate' was to figure out what these provide. 
`last_run_penultimate_status (status on second to last ingested ccr)`
`last_scan_penultimate_status (status on second to last ingested scan)`
